sdk/python/kfp/_client.py

```diff
@@ -277,9 +285,22 @@ def set_user_namespace(self, namespace):
       namespace: kubernetes namespace the user has access to.
     """
     self._context_setting['namespace'] = namespace
+    if not os.path.exists(os.path.dirname(Client.LOCAL_KFP_CONTEXT)):
+        os.makedirs(os.path.dirname(Client.LOCAL_KFP_CONTEXT))
     with open(Client.LOCAL_KFP_CONTEXT, 'w') as f:
       json.dump(self._context_setting, f)
 
+  def get_kfp_healthz(self):
+    """Get healthz info of KFP deployment.
+
+    Returns:
+      response: json formatted response from the healtz endpoint.
+    """
+    healthz_api = 'http://' + Client.IN_CLUSTER_DNS_NAME.format('kubeflow') + '/' + Client.HEALTH_PATH
```

Comment:
We have `host` passed in as an optional parameter. We should use it or the resolved host instead of always using IN_CLUSTER_DNS_NAME

Reply:
@Bobgy I have changed it to `self._existing_config.host`, which I believe is the correct to grab the resolved host.